src/functionaltests/java/com/ericsson/ei/subscriptions/authentication/AuthenticationSteps.java

```diff
@@ -0,0 +1,103 @@
+package com.ericsson.ei.subscriptions.authentication;
+
+import com.ericsson.ei.controller.AuthControllerImpl;
+import com.ericsson.ei.utils.FunctionalTestBase;
+import com.ericsson.ei.utils.TestLDAPInitializer;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.apache.commons.io.FileUtils;
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.apache.tomcat.util.codec.binary.StringUtils;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.io.File;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Ignore
+@AutoConfigureMockMvc
+@ContextConfiguration(initializers = TestLDAPInitializer.class)
+public class AuthenticationSteps extends FunctionalTestBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationSteps.class);
+    private static final String SUBSCRIPTION = "src/functionaltests/resources/subscription_single.json";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private AuthControllerImpl authController;
+
+    private MvcResult mvcResult;
+
+    private String requestBody;
+
+    @Given("^LDAP is activated$")
+    public void ldap_is_activated() throws Throwable {
+        String responseBody = new JSONObject().put("security", true).toString();
+        mockMvc.perform(MockMvcRequestBuilders.get("/auth")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isOk())
+            .andExpect(content().string(responseBody))
+            .andReturn();
```

Comment:
Do we need to log what failed or do mockMvc log a good detailed description on what failed when result is not as Expected ?

Reply:
This comment applies to all test steps.